featuremanagement/azuremonitor/_send_telemetry.py

```diff
@@ -53,50 +55,53 @@ def publish_telemetry(evaluation_event: EvaluationEvent) -> None:
     """
     if not HAS_AZURE_MONITOR_EVENTS_EXTENSION:
         return
-    event: Dict[str, Optional[str]] = {}
-    if not evaluation_event.feature:
+
+    feature = evaluation_event.feature
+
+    if not feature:
         return
-    event[FEATURE_NAME] = evaluation_event.feature.name
-    event[ENABLED] = str(evaluation_event.enabled)
-    event["Version"] = EVALUATION_EVENT_VERSION
+
+    event: Dict[str, Optional[str]] = {
+        FEATURE_NAME: feature.name,
+        ENABLED: str(evaluation_event.enabled),
+        "Version": EVALUATION_EVENT_VERSION,
+    }
+
+    reason = evaluation_event.reason
+    variant = evaluation_event.variant
 
     # VariantAllocationPercentage
-    if evaluation_event.reason and evaluation_event.reason != VariantAssignmentReason.NONE:
-        if evaluation_event.variant:
-            event[VARIANT] = evaluation_event.variant.name
-        event[REASON] = evaluation_event.reason.value
-
-        if evaluation_event.reason == VariantAssignmentReason.DEFAULT_WHEN_ENABLED:
-            allocation_percentage = 0
-
-            if evaluation_event.feature.allocation and evaluation_event.feature.allocation.percentile:
-                for allocation in evaluation_event.feature.allocation.percentile:
-                    if (
-                        evaluation_event.variant
-                        and allocation.variant == evaluation_event.variant.name
-                        and allocation.percentile_to
-                    ):
-                        allocation_percentage += allocation.percentile_to - allocation.percentile_from
+    if reason and reason != VariantAssignmentReason.NONE:
+        if variant:
+            event[VARIANT] = variant.name
+        event[REASON] = reason.value
+
+        allocation_percentage = 0
+        if (
+            reason == VariantAssignmentReason.DEFAULT_WHEN_ENABLED
+            and feature.allocation
+            and feature.allocation.percentile
+        ):
+            for allocation in feature.allocation.percentile:
+                if variant and allocation.variant != variant.name and allocation.percentile_to:
```

Comment:
I am confused. Should it be allocation.variant == variant.name?

Reply:
I still feel weird to only check percentile_to here. percentile_from is also used below. I think if we need to validate it, them both of from and to should be validated.

Reply:
For the first part I think I swapped the value in my mind forgetting we subtract later. @rossgrambo, what do you think.

Reply:
Also @rossgrambo can you don't check this, as the unit test now breaks with the ==.

Reply:
The if statement (just line 86) should be removed. All we want here is the total allocated percentiles. We don't care what variant is allocated in any of them. For someone using the portal- this will always add up to 100.

Reply:
Just pushed a fix for this.